docs/tutorial.md

```diff
@@ -0,0 +1,198 @@
+- Install [kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/)
+
+## Overview
+
+We will create a sample project to let you know how it works and this sample will:
+
+- Create a `memcached-operator` using below mentioned steps. For more infomation refer to this [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/)
+- Generate the manifests using `config/default`
+- Comment out `ClusterRole` and `ClusterRoleBinding`. 
+- Create a `ClusterRole` template in a `ScopeTemplate` and associate it with a `ScopeInstance`
+- Run the operator, the `memcached-operator` will start complaining about `RBAC`.
+- Run the `oria-operator`, it will create `ClusterRole` and `ClusterRoleBinding`. This will make sure that `memcached-operator` will not complain about `RBAC`. 
+
+### Install and run oria-operator
+
+Clone the [github repo](https://github.com/operator-framework/oria-operator.git)
+
+```
+git clone https://github.com/operator-framework/oria-operator.git
+```
+
+Then, run the `oria-operator` with below commands on the root of the directory.
+
+```
+$ make install run
+```
+
+The above process runs in the foreground. You will need to open a new terminal window/tab to continue with the rest of the tutorial.
+
+### Create a new project
+
+Use the [Operator SDK](https://sdk.operatorframework.io/docs/installation/) CLI to create a new memcached-operator from the [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/).
+
+Create a directory for the memcached operator:
+
+```
+mkdir -p <directory>
+```
+
+Navigate into the newly created directory and initialize the project with:
+
+```sh
+operator-sdk init --domain example.com --repo github.com/example/memcached-operator
+```
+
+### Create a new API and Controller 
+
+```sh
+$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource --controller
+```
+
+Define the API for the Memcached Custom Resource(CR) by modifying the Go type definitions at `api/v1alpha1/memcached_types.go` to have the following spec and status:
+
+```
+// MemcachedSpec defines the desired state of Memcached
+type MemcachedSpec struct {
+	//+kubebuilder:validation:Minimum=0
+	// Size is the size of the memcached deployment
+	Size int32 `json:"size"`
+}
+
+// MemcachedStatus defines the observed state of Memcached
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+```
+
+### Now create docker image and push to the dockerhub
+
+Your Makefile composes image tags either from values written at project initialization or from the CLI. In particular, IMAGE_TAG_BASE lets you define a common image registry, namespace, and partial name for all your image tags. Update this to another registry and/or namespace if the current value is incorrect. Afterwards you can update the IMG variable definition like so:
+
+```
+-IMG ?= controller:latest
++IMG ?= $(IMAGE_TAG_BASE):$(VERSION)
+```
```

Comment:
A follow-on PR, we can remove this because we are overriding the `IMG` value with the command below. That is perfectly acceptable. Keep it here for now.

Reply:
updated this too.